app/scripts/modules/core/src/pipeline/config/actions/templateJson/ShowPipelineTemplateJsonModal.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { cloneDeep } from 'lodash';
+import { IModalComponentProps, JsonEditor } from 'core/presentation';
+import { IPipeline, IPipelineTemplateV2 } from 'core/domain';
+import { CopyToClipboard, noop, JsonUtils } from 'core/utils';
+import { PipelineTemplateV2Service } from 'core/pipeline/config/templates/v2/pipelineTemplateV2.service';
+import './showPipelineTemplateJsonModal.less';
+
+const commandCopy = 'Copy "spin pipeline-templates save" command to clipboard';
+
+export interface IShowPipelineTemplateJsonModalProps extends IModalComponentProps {
+  ownerEmail: string;
+  pipeline: IPipeline;
+}
+
+export interface IShowPipelineTemplateJsonModalState {
+  template: IPipelineTemplateV2;
+}
+
+export class ShowPipelineTemplateJsonModal extends React.Component<
+  IShowPipelineTemplateJsonModalProps,
+  IShowPipelineTemplateJsonModalState
+> {
+  public static defaultProps: Partial<IShowPipelineTemplateJsonModalProps> = {
+    dismissModal: noop,
+  };
+
+  constructor(props: IShowPipelineTemplateJsonModalProps) {
+    super(props);
+
+    const template = PipelineTemplateV2Service.createPipelineTemplate(cloneDeep(props.pipeline), props.ownerEmail);
+    this.state = { template };
+  }
+
+  private onChange = (e: React.ChangeEvent<HTMLInputElement>, property: string) =>
+    this.setState({
+      template: {
+        ...this.state.template,
+        metadata: {
+          ...this.state.template.metadata,
+          [property]: e.target.value,
+        },
+      },
+    });
+
+  public render() {
+    const { dismissModal } = this.props;
+    const { template } = this.state;
+    const templateStr = JsonUtils.makeSortedStringFromObject(template);
+
+    return (
+      <div className="flex-fill">
+        <div className="modal-header">
+          <h3>Pipeline Template JSON</h3>
+        </div>
+        <div className="modal-body flex-fill">
+          <p>The JSON below represents the derived template from the pipeline configuration in its persisted state.</p>
```

Comment:
Suggest changing the language here to be a bit less technical. Something like "The JSON below is the templated version of your pipeline. Save it by copy/pasting to the spin CLI tool."

Reply:
👍